### ISSUE TITLE
Out of tree build

### ISSUE THREAD

**Issue by hommeabeil** [meta-erlang/meta-erlang#112]
I just discover we can build an erlang project with rebar3 out tree, well mostly. Are you aware of the environment variable `REBAR_BASE_DIR` [link](https://github.com/erlang/rebar3/blob/master/src/rebar3.erl#L154) ? It would allow us to output all the binary into  `${B}`. All  I need to do in my recipe is to do something like this

```
B = "${WORKDIR}/build"
export REBAR_BASE_DIR="${B}"

do_compile_prepend() {
     # This is needed since the default is to be in ${B}. In fack all the task which invoke rebar3 need its.
    cd ${S}
}
# Other prepend need the cd to
```

I think this can be useful, mostly for the SDK. I am opening an issue to have your input and to know if their is a reason to not use it. I would be able to make the contribution if you think this would work.

**Comment by joaohf** [comment]
Hello @hommeabeil ,

I didn't know about that variable. Indeed as described here https://github.com/erlang/rebar3/issues/2104 it could use as you said. I don't any downside in use that in a recipe. I'll do some test to see how it could fit better here (mainly for SDK and when making releases).

Do you have any workflow in mind ?

Thanks.

**Comment by hommeabeil** [comment]
The workflow I mostly want to achieve is to be able to use the SDK with an already clone repo (`devtool modify -n <recipe> <path>`). Currently if I am not mistaken, this will try to build in the source and might fail if the _build already contains some artefacts. So I need to delete de _build directory and then redo a `devtool build <recipe>`. When I finish with the SDK I need to clean the _build again if I want to build without the SDK.

In short I would like to keep the build artefacts from the SDK outside of my cloned repository. Other than that, from the yocto build system everything works well.

**Comment by joaohf** [comment]
Hello @hommeabeil 

Now I fully understood the picture and in fact it is really nice. Solving some old issues. Mainly when developing.

What do you think about this one: https://github.com/meta-erlang/meta-erlang/compare/out-of-tree ?

**Comment by hommeabeil** [comment]
This look like what I had in mind. I will give it a try this week.
